cli/src/semgrep/output.py

```diff
@@ -118,11 +119,40 @@ def _build_time_json(
     )
 
 
+# This class is the internal representation of OutputSettings below.
+# Since it is internal it can change as much as necesarry to make
+# typechecking more accurate and enforce invariants.
```

Comment:
would be nice to explain here an example of invariant, that is not enforced by OutputSettings but is by NormalizedOutputSettings.
Maybe just say like "for example, outputs is now a Dict[...] instead of Optional[Dict[...]]